app/src/main/java/com/rkm/tasky/sync/repository/implementation/SyncUpdateEventRepositoryImpl.kt

```diff
@@ -38,24 +40,28 @@ class SyncUpdateEventRepositoryImpl @Inject constructor(
         }
 
     override suspend fun upsertEvent(event: UpdateEventSyncRequest) = withContext(dispatcher) {
-        syncDataSource.upsertSyncItem(
-            SyncEntity(
-                action = SyncUserAction.UPDATE,
-                item = SyncItemType.EVENT,
-                itemId = event.id
-            )
-        )
-        attendeesDataSource.upsertAttendees(event.asSyncAttendeeEntity())
-        deletedPhotosDataSource.upsertPhotoKeys(event.asSyncDeletePhotoEntity())
-        uploadPhotoDataSource.upsertPhotos(event.asSyncUploadPhotosEntity())
-        eventDataSource.upsertEvent(event.asSyncUpdateEventEntity())
+        listOf(
+            launch { syncDataSource.upsertSyncItem(
+                SyncEntity(
+                    action = SyncUserAction.UPDATE,
+                    item = SyncItemType.EVENT,
+                    itemId = event.id
+                )
+            ) },
+            launch { attendeesDataSource.upsertAttendees(event.asSyncAttendeeEntity()) },
+            launch { deletedPhotosDataSource.upsertPhotoKeys(event.asSyncDeletePhotoEntity()) },
+            launch { uploadPhotoDataSource.upsertPhotos(event.asSyncUploadPhotosEntity()) },
+            launch { eventDataSource.upsertEvent(event.asSyncUpdateEventEntity()) }
+        ).joinAll()
```

Comment:
If you have to manipulate that many parts of your DB, better make use of transactions. Otherwise, if one single call of these 5 calls here fails, you'll end up with an inconsistent DB/Event. A transaction will make sure to only apply these changes if all calls have succeeded.

In Room, you can create a transaction like this (just as an example, needs to be adjusted to your use case):
```
@Dao
interface AgendaDao {

    @Transaction
    suspend fun updateAgenda(db: AgendaDatabase, agendaItems: List<AgendaItem>) {
        db.apply {
            eventDao.deleteAllEvents()
            attendeeDao.deleteAllAttendees()
            taskDao.deleteAllTasks()
            reminderDao.deleteAllReminders()

            val events = agendaItems.filterIsInstance<AgendaItem.Event>()
            val tasks = agendaItems.filterIsInstance<AgendaItem.Task>()
            val reminders = agendaItems.filterIsInstance<AgendaItem.Reminder>()
            events.forEach {
                eventDao.insertEvent(it.toEventEntity())
                it.attendees.forEach {
                    attendeeDao.insertAttendee(it.toAttendeeEntity)
                }
            }
            tasks.forEach {
                taskDao.insertTask(it.toTaskEntity())
            }
            reminders.forEach {
                reminderDao.insertReminder(it.toReminderEntity())
            }
        }
    }
}
```
Whenever you need to execute multiple DB calls that should only be applied to the DB if they all succeed, you need something like this.